plugins/metadata/services/MetadataService.php

```diff
@@ -78,6 +78,15 @@ function addAction($metadataProfileId, $objectType, $objectId, $xmlData)
 		}
 		
 		$objectType = kPluginableEnumsManager::apiToCore('MetadataObjectType', $objectType);
+
+		$limitEntry = $this->getKs()->getSetLimitEntry();
+		$peer = kMetadataManager::getObjectPeer($objectType);
```

Comment:
No need to instantiate the peer and query the DB for the entry if $limitEntry is not set.